### PR TITLE
Add exception handling when 2 copies of the app access DB simultaneously.

### PR DESCRIPTION
**Pull request by vvechkanov**
* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests if this modifies the Windows code?
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Add exception handling when 2 copies of the app access DB simultaneously.
 
## Related PRs or issues

[AB#82171](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/82171)

## Misc

Add what's missing, notes on what you tested, additional thoughts or questions.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,6 +16,7 @@ ___
 #### WPF/WinForms
 
 * **[Fix]** Fix exception handling if App Center is disabled during sending logs.
+* **[Fix]** Fix exception handling when the app tries to access a locked database.
```
Comment (MatkovIvan) [inline CHANGELOG.md:19]: This version is already released, please move it to the right one

File: Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs [hunk 2]
```diff
@@ -516,6 +516,128 @@ public async Task DisableChannelDuringSendingLogs(int countLogs)
...
+                        e.SetObserved();
+                    }
+                };
+                var storage = new Mock<IStorage>();
+                storage.Setup(s => s.DeleteLogs(It.IsAny<string>())).Throws(storageException);
+                storage.Setup(s => s.GetLogsAsync(It.IsAny<string>(), It.IsAny<int>(), It.IsAny<List<Log>>()))
+                    .Callback((string channelName, int limit, List<Log> logs) => logs.Add(log))
+                    .Returns(() => Task.FromResult("test-batch-id"));
+
+
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:543]:
```suggestion
```

File: Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs [hunk 3]
```diff
@@ -516,6 +516,128 @@ public async Task DisableChannelDuringSendingLogs(int countLogs)
...
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:584]:
```suggestion
```

File: Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs [hunk 4]
```diff
@@ -516,6 +516,128 @@ public async Task DisableChannelDuringSendingLogs(int countLogs)
...
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:626]:
```suggestion
```

File: Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs [hunk 5]
```diff
@@ -516,6 +516,128 @@ public async Task DisableChannelDuringSendingLogs(int countLogs)
...
+                // Prepare data.
+                var appSecret = Guid.NewGuid().ToString();
+                Channel channel = new Channel(ChannelName, MaxLogsPerBatch, new TimeSpan(), MaxParallelBatches,
+                    appSecret, _mockIngestion.Object, storage.Object);
+
+                // Disable and enable channel.
+                channel.FailedToSendLog += (sender, args) => { };
```
Comment (MatkovIvan) [inline Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:591]: Why do you need it?

Comment (AnastasiaKubova) [reply Tests/Microsoft.AppCenter.Test.Windows/Channel/ChannelTest.cs:591]: It need to cover [this](https://github.com/microsoft/appcenter-sdk-dotnet/pull/1429/files#diff-f5d602677f469ef250175751395c23dcR365) catch block.

File: SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/Channel.cs [hunk 6]
```diff
@@ -269,6 +269,10 @@ public async Task ClearAsync()
             {
                 AppCenterLog.Warn(AppCenterLog.LogTag, "The Clear operation has been canceled");
             }
+            catch (StorageException e)
+            {
+                AppCenterLog.Warn(AppCenterLog.LogTag, $"Could not delete logs with error: ", e);
```
Comment (MatkovIvan) [inline SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Channel/Channel.cs:274]: Please remove `$` where it isn't needed

**Pull request merged**